src/sas/sascalc/pr/p_invertor.py

```diff
@@ -0,0 +1,663 @@
+        """
+        q = np.float64(q)
+        q = np.atleast_1d(q)
+        pars = np.float64(pars)
+
+
+        npts = 21
+        iq_val = calc.iq_smeared(pars, q, self.d_max, self.slit_height, self.slit_width, npts)
+        #If q was a scalar
+        if iq_val.shape[0] == 1:
+            return np.asscalar(iq_val)
+        return iq_val
+
+    def pr(self, pars, r):
+        """
+        Function to call to evaluate P(r).
+
+        :param pars: c-parameters.
+        :param r: r-value to evaluate P(r) at.
+
+        :return: P(r)
+        """
+        r = np.float64(r)
+        pars = np.float64(pars)
+        pars = np.atleast_1d(pars)
+        r = np.atleast_1d(r)
+        pr_val = calc.pr(pars, self.d_max, r)
+        if len(pr_val) == 1:
+            #scalar
+            return np.asscalar(pr_val)
+        return pr_val
+
+    def get_pr_err(self, pars, pars_err, r):
+        """
+        Function to call to evaluate P(r) with errors.
+
+        :param pars: c-parameters.
+        :param pars_err: pars_err.
+        :param r: r-value.
+
+        :return: (P(r), dP(r))
+        """
+
+        pars = np.atleast_1d(np.float64(pars))
+        r = np.atleast_1d(np.float64(r))
+        if pars_err is None:
+            return calc.pr(pars, self.d_max, r), 0.0
+        else:
+            pars_err = np.float64(pars_err)
+            return calc.pr_err(pars, pars_err, self.d_max, r)
+
+    def is_valid(self):
+        """
+        Check the validity of the stored data.
+
+        :return: Returns the number of points if it's all good, -1 otherwise.
+        """
+        if self.npoints == self.ny and self.npoints == self.nerr:
+            return self.npoints
+        else:
+            return -1
+
+    def basefunc_ft(self, d_max, n, q):
+        """
+        Returns the value of the nth Fourier transformed base function.
+
+        :param d_max: d_max.
+        :param n: n.
+        :param q: q, scalar or vector.
+
+        :return: nth Fourier transformed base function, evaluated at q.
+        """
+        d_max = np.float64(d_max)
+        n = int(n)
+        q = np.float64(q)
+        q = np.atleast_1d(q)
+        ortho_val = calc.ortho_transformed(q, d_max, n)
+
+        if ortho_val.shape[0] == 1:
+            #If the q input was scalar.
+            return np.asscalar(ortho_val)
+        return ortho_val
+
+    def oscillations(self, pars):
+        """
+        Returns the value of the oscillation figure of merit for
+        the given set of coefficients. For a sphere, the oscillation
+        figure of merit is 1.1.
+
+        :param pars: c-parameters.
+        :return: oscillation figure of merit.
+        """
+        nslice = 100
+        pars = np.float64(pars)
+        pars = np.atleast_1d(pars)
+
+        oscill = calc.reg_term(pars, self.d_max, nslice)
+        norm = calc.int_pr_square(pars, self.d_max, nslice)
+        ret_val = np.sqrt(oscill/norm) / np.arccos(-1.0) * self.d_max
```

Comment:
Use `np.pi` instead of `np.arccos(-1.0)`.

Reply:
Done.